### DISCUSSION TITLE
Request requires valid admin authorization token to be set. Err 401

### DISCUSSION THREAD

**Discussion by cheyhauf** [pocketbase/pocketbase#2294] [Q&A] [answered]
Hello, I'm currently trying to use pocketbase as a database apart from a node express ejs, so SRR app. All seems to work fine in development, and I'm able to access the pocketbase web UI behind an nginx reverse proxy on my domain as its running as a service from the minimal deployment on a digital ocean droplet, ubuntu 22. When I try to authenticate as an admin, I'm met with an error,

ClientResponseError 401: The request requires valid admin authorization token to be set.
    at new ClientResponseError
 {
    url: 'https://mydomain.com/PBA/api/admins/auth-with-password',
    status: 401,
    response: {
    code: 401,
    message: 'The request requires valid admin authorization token to be set.',
    data: {}
  },
  isAbort: false,
  originalError: null
}

All through development and testing on my local device, I was able to authenticate with no trouble, as the .env variables are making it to the pb.admins.authWithPassword. The admin exists with the login on the ubuntu instance. 

```
require('dotenv').config();
const PocketBase = require('pocketbase/cjs');
const pb = new PocketBase('http://mydomain.com/PBA');

const auth = async () => {
    console.log('DB_USER:', process.env.DB_USER); // Exists
    console.log('DB_PASSWORD:', process.env.DB_PASSWORD); // Also Exists

    const authData = await pb.admins.authWithPassword(process.env.DB_USER, process.env.DB_PASSWORD);
};
auth();
```

Pocket base portion of nginx config

```
location /PBA {
        proxy_set_header Connection '';
        proxy_http_version 1.1;
        proxy_read_timeout 360s;

        proxy_set_header Host $host;
        proxy_set_header X-Real-IP $remote_addr;
        proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
        proxy_set_header X-Forwarded-Proto $scheme;

        rewrite /PBA/(.*) /$1  break;

        proxy_pass http://IP;
    }
```

I am wondering if it is a simple misconfiguration with nginx or another piece I'm overlooking but I'm quite stumped and any insight or guidance is appreciated!

**Comment by ganigeorgiev** [comment]
The error message is definitely strange as the `/auth-with-password` is public and I don't see how 401 will be returned.

Could you confirm that `http://mydomain.com/PBA/_/` correctly resolves the Admin UI? Internally the Admin UI also uses the JS SDK, so if it is working there, it also should work with your node application.

Could you provide a minimal reprodible repo or at least a little more code details when you call `auth()` (note that it is async and if you are not awaiting it and sending another request with it, it is not guaranteed that the `auth()` call will be processed first)?

**Reply by cheyhauf** [reply]
Yep, accessing http://mydomain.com/PBA/_/ will resolve the admin UI and I'm able to manipulate records and admins accordingly upon logging in, although it is always redirected to https, not sure if this is notable. I'm able to reproduce this error message when I curl the url from the ubuntu instance it's running on, trying to access the url from my dev machine, and as little code as this on the node instance on the server 

```
const express = require('express');
const app = express();
const path = require('path');

app.use(express.static(__dirname + '/dist/'));


const PocketBase = require('pocketbase/cjs');
const pb = new PocketBase('http://domain.com/PBA');

const auth = async () => {
    const authData = await pb.admins.authWithPassword('foobar@foo.bar', 'foobar');
};
auth();

app.get('/*', function(req,res) {
    res.sendFile(path.join(__dirname+'/dist/index.html'));
    }
);

app.listen(1234, () => {
    console.log("Server started on port 1234");
});
```

**Reply by ganigeorgiev** [reply, marked as answer]
> although it is always redirected to https, not sure if this is notable. 

Could you try changing the PocketBase initialization address in your node to use the https version? Maybe it is not following the redirect, I'm not sure. I'm not able to reproduce the error locally.

**Reply by cheyhauf** [reply]
Sure enough, changing the systemd service and to 

`pocketbase serve --https="dropletIP:MyPort" `

and

`const pb = new PocketBase('https://domain.com/PBA');`

Solved this issue, I appreciate the guidance and timely replies! I also really enjoy using your software and excited to see where it goes in the future!
